unstructured_inference/models/unstructuredmodel.py

```diff
@@ -55,6 +63,116 @@ def __call__(self, x: Image) -> List[LayoutElement]:
+            if first:
+                # We get only the elements which intersected
+                indices_to_check = np.where(row)[0]
+                # Delete the first element, since it will always intersect with itself
+                indices_to_check = indices_to_check[indices_to_check != i]
+                if len(indices_to_check) == 0:
+                    continue
+                if len(indices_to_check) > 1:  # sort by iom
+                    iom_to_check = [
+                        (j, first.intersection_over_minimum(elements[j]))
+                        for j in indices_to_check
+                        if elements[j] is not None
+                    ]
+                    iom_to_check.sort(
+                        key=lambda x: x[1],
+                        reverse=True,
+                    )  # sort elements by iom, so we first check the greatest
+                    indices_to_check = [x[0] for x in iom_to_check]  # type:ignore
+            for j in indices_to_check:
+                if i != j and elements[j] is not None and elements[i] is not None:
+                    second = elements[j]
+                    intersection = first.intersection(
+                        second,
+                    )  # we know it does, but need the region
+                    first_inside_second = first.is_in(second)
+                    second_inside_first = second.is_in(first)
+
+                    if first_inside_second and not second_inside_first:
+                        elements[i] = None  # type:ignore
+                    elif second_inside_first and not first_inside_second:
+                        # delete second element
+                        elements[j] = None  # type:ignore
+                    elif intersection:
+                        iom = first.intersection_over_minimum(second)
+                        if iom < iom_to_merge:  # small
+                            separate(first, second)
+                            # The rectangle could become too small, which is a
+                            # good size to delete?
+                        else:  # big
+                            # merge
+                            if first.area > second.area:
+                                grow_region_to_match_region(first, second)
+                                elements[j] = None  # type:ignore
+                            else:
+                                grow_region_to_match_region(second, first)
+                                elements[i] = None  # type:ignore
+
+        elements = [e for e in elements if e is not None]
+        return elements
+
+    @staticmethod
+    def clean_type(elements: List[LayoutElement], type="Table") -> List[LayoutElement]:
+        """After this function, the list of elements will not contain any element inside
+        of the type specified"""
+        target_elements = [e for e in elements if e.type == "Table"]
+        other_elements = [e for e in elements if e.type != "Table"]
```

Comment:
let's also rename this variable to `type_to_clean` so we don't override a reserved word `type`
```suggestion
        target_elements = [e for e in elements if e.type == type_to_clean]
        other_elements = [e for e in elements if e.type != type_to_clean]
```

Reply:
OMG, right!